src/main/java/org/jabref/gui/collab/entrychange/EntryWithPreviewAndSourceDetailsView.java

```diff
@@ -0,0 +1,74 @@
+package org.jabref.gui.collab.entrychange;
+
+import java.io.IOException;
+import java.io.StringWriter;
+
+import javafx.scene.control.Tab;
+import javafx.scene.control.TabPane;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.Globals;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.collab.DatabaseChangeDetailsView;
+import org.jabref.gui.preview.PreviewViewer;
+import org.jabref.gui.theme.ThemeManager;
+import org.jabref.logic.bibtex.BibEntryWriter;
+import org.jabref.logic.bibtex.FieldWriter;
+import org.jabref.logic.bibtex.FieldWriterPreferences;
+import org.jabref.logic.exporter.BibWriter;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.util.OS;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.database.BibDatabaseMode;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.preferences.PreferencesService;
+
+import org.fxmisc.flowless.VirtualizedScrollPane;
+import org.fxmisc.richtext.CodeArea;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class EntryWithPreviewAndSourceDetailsView extends DatabaseChangeDetailsView {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EntryWithPreviewAndSourceDetailsView.class);
+
+    public EntryWithPreviewAndSourceDetailsView(BibEntry entry, BibDatabaseContext bibDatabaseContext, DialogService dialogService, StateManager stateManager, ThemeManager themeManager, PreferencesService preferencesService) {
+        PreviewViewer previewViewer = new PreviewViewer(bibDatabaseContext, dialogService, stateManager, themeManager);
+        previewViewer.setLayout(preferencesService.getPreviewPreferences().getSelectedPreviewLayout());
+        previewViewer.setEntry(entry);
+
+        CodeArea codeArea = new CodeArea();
+        codeArea.setId("bibtexcodearea");
+        codeArea.setWrapText(true);
+        codeArea.setDisable(true);
```

Comment:
I would use `SourceTab` for reusability. You can also add a new method to switch between read-only and editable mode

Reply:
I first thought the same, but then I saw that it directly extends from the EntryEditor Tab and contain a lot of bindings etc that are not needed. So I opted for the simplest solution.